go/vt/vtgate/planbuilder/testdata/select_cases.json

```diff
@@ -2043,6 +2043,159 @@
       ]
     }
   },
+  {
```

Comment:
nitpick: it'd be great to test these in an end to end test as well

Reply:
I agree, adding to the e2e test would be nice.

Reply:
Done